src/dmd/mtype.d

```diff
@@ -7233,7 +7233,7 @@ ScopeRef buildScopeRef(bool returnByRef, StorageClass stc) pure nothrow @nogc @s
     ScopeRef result;
     final switch (stc & (STC.ref_ | STC.scope_ | STC.return_))
     {
-        case STC.return_:
+        case STC.return_:              // can occur in case test/compilable/testsctreturn.d
```

Comment:
Looking at the test case, this actually looks like a bug related to https://issues.dlang.org/show_bug.cgi?id=20149. `inout` puts the compiler in an inconsistent state by adding storage class `return` without `scope` or `ref`, and now instead of an assertion failure, the compiler turns a blind eye.

Reply:
Fixing that problem is one for another PR. Please file a bugzilla for it.

Reply:
The link I gave is a bugzilla link: https://issues.dlang.org/show_bug.cgi?id=20149